### PR TITLE
Whats on

### PR DESCRIPTION
**Pull request by gestchild**
References #1796

This is a pretty large PR (sorry) although lots such as the focus-trap.js/segmented-control.js  was already reviewed here: #1938

The main things to concentrate on here will be related to the service infrasctructure, the what's on templates/styling and whats-on/controllers.js

N.B.  browser that don't support grids currently show the promos in a vertical list. Looking at doing a better fallback based on flex-box, but will put in another PR

## Type
✨ Feature  

## Value
Adds the whats on service/landing page

## Screenshot
localhost_3003_whats-on__f 5bdates 5d 2018-01-08 7c2018-04-08

### UI component only
- [x] A11y tested: `npm run test:accessibility <URL>`
- [x] Structured data tested: https://search.google.com/structured-data/testing-tool 
- [x] Browser tested: `npm run test:browsers <URL>`
- [x] Works without JS in the client
- [ ] Relevant tracking/monitoring has been considered

### CODE REVIEW

File: client/scss/utilities/_mixins.scss [hunk 1]
```diff
@@ -94,7 +128,14 @@
 @mixin css-grid-cells($columns, $key) {
   @for $i from 1 through $columns {
     .css-grid__cell--#{$key}#{$i} {
-      grid-column: span #{$i};
+      max-width: ($i / $columns) * 100%;
+      margin-bottom: 3 * $v-spacing-unit;
```
Comment (jamesgorrie) [inline client/scss/utilities/_mixins.scss]: Are we 100% defining this on every cell that is used throughout the site?
Just want to make sure we don't undo the good work done on spacingClasses.

File: client/scss/utilities/_root-scope-classes.scss [hunk 2]
```diff
@@ -367,3 +367,11 @@
 .overflow-hidden {
   overflow: hidden;
 }
+
+// To be applied to white Wellcome Bold to prevent
+// it looking too chunky on dark backgrounds
+.smooth-font {
+  -webkit-font-smoothing: antialiased;
```
Comment (jamesgorrie) [inline client/scss/utilities/_root-scope-classes.scss]: Any reason not to add `font-smoothing`?

File: events/app/views/pages/events.njk [hunk 3]
```diff
@@ -15,42 +15,52 @@
...
+        {% componentV2 'divider', null, {'keyline': true, 'pumice': true} %}
       </div>
     </div>
   </div>
+  {% endif %}
+
+  <div class="css-grid__container {{ {s:4} | spacingClasses({padding: ['top']}) }} ">
+    <div class="css-grid">
+      {% for event in paginatedEvents.results %}
+        <div class="{{ {s: 12, m: 6, l:3, xl:3} | cssGridClasses }}">
```
Comment (jamesgorrie) [inline events/app/views/pages/events.njk]: any reason to call it  `cssGridClasses` vs `gridClasses`?

File: whats-on/terraform/services.tf [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+  server_error_alarm_topic_arn = "${module.alb_server_error_alarm.arn}"
+  client_error_alarm_topic_arn = "${module.alb_client_error_alarm.arn}"
+
+  # These account for the 128 mem and CPU the nginx container use
+  # 995 is how much memmory is left once docker is running
+  cpu                      = "384" # (1024/2) - 128
+  memory                   = "369" # (995/2) - 128
+  primary_container_port   = "80"
+  secondary_container_port = "3002"
```
Comment (jamesgorrie) [inline whats-on/terraform/services.tf]: Have we updated the cluster to allow for this?

File: whats-on/terraform/services.tf [hunk 5]
```diff
@@ -0,0 +1,34 @@
+module "events" {
```
Comment (jamesgorrie) [inline whats-on/terraform/services.tf]: Naming?

File: whats-on/terraform/services.tf [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  client_error_alarm_topic_arn = "${module.alb_client_error_alarm.arn}"
+
+  # These account for the 128 mem and CPU the nginx container use
+  # 995 is how much memmory is left once docker is running
+  cpu                      = "384" # (1024/2) - 128
+  memory                   = "369" # (995/2) - 128
+  primary_container_port   = "80"
+  secondary_container_port = "3002"
+
+  path_pattern = "/whats-on/*"
```
Comment (jamesgorrie) [inline whats-on/terraform/services.tf]: `/whats-on*`?

**Comment by jamesgorrie** [comment]
Would merging #1938 make it easer to review?

File: whats-on/app/controllers.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+  restaurantPromo,
+  spiritBoothPromo
+} from '../../server/data/daily-promos';
+const {createPageConfig} = model;
+const {
+  getExhibitionAndEventPromos
+} = prismic;
+
+export async function renderWhatsOn(ctx, next) {
+  const { 'f[dates]': timeSpan } = ctx.query;
```
Comment (jamesgorrie) [inline whats-on/app/controllers.js]: Why this as a query?

File: whats-on/app/views/partials/date_range.njk [hunk 8]
```diff
@@ -0,0 +1,9 @@
+{% set fromDate = exhibitionAndEventPromos.dates.queriedDates[0] %}
+{% set toDate = exhibitionAndEventPromos.dates.queriedDates[1] %}
+{% set isMoreThanOneDay =  fromDate !== toDate %}
+
+<p class="{{- [
+  {s:'HNM4', m:'HNM4'} | fontClasses,
+  {s:0} | spacingClasses({margin: ['top']}),
+  {s:1} | spacingClasses({margin: ['bottom']})
+].join(' ') -}}"><time datetime="{{ fromDate }}">{{ fromDate | formatDate }}</time>{%- if isMoreThanOneDay -%}&ndash;<time datetime="{{ toDate }}">{{ toDate | formatDate }}</time>{%- endif -%}</p>
```
Comment (jamesgorrie) [inline whats-on/app/views/partials/date_range.njk]: What is the partials folder for?

**Comment by gestchild** [comment]
@jamesgorrie I'm splitting this up into smaller PRs now; want to hold off reviewing for now?

**Comment by gestchild** [comment]
splitting this into more manageable chuncks

**Pull request closed without merging**